base/applications/cmdutils/attrib/attrib.c

```diff
@@ -77,217 +59,260 @@ ErrorMessage(
         ConPrintf(StdErr, L"  %s\n", pszMsg);
 }
 
-/* Returns TRUE if anything is printed, FALSE otherwise */
+
+/**
+ * @brief   Displays attributes for the given file.
+ * @return  Always TRUE (success).
+ **/
 static
 BOOL
-PrintAttribute(
-    LPWSTR pszPath,
-    LPWSTR pszFile,
-    BOOL   bRecurse,
-    BOOL   bDirectories)
+PrintAttributes(
+    _In_ PWIN32_FIND_DATAW pFindData,
+    _In_ PCWSTR pszFullName,
+    _Inout_opt_ PVOID Context)
 {
-    WIN32_FIND_DATAW findData;
-    HANDLE hFind;
-    WCHAR  szFullName[MAX_PATH];
-    LPWSTR pszFileName;
-    BOOL   bFound = FALSE;
-    BOOL   bIsDir;
-    BOOL   bExactMatch;
-    DWORD  Error;
-
-    /* prepare full file name buffer */
-    wcscpy(szFullName, pszPath);
-    pszFileName = szFullName + wcslen(szFullName);
-
-    /* display all subdirectories */
-    if (bRecurse)
-    {
-        /* append *.* */
-        wcscpy(pszFileName, L"*.*");
-
-        hFind = FindFirstFileW(szFullName, &findData);
-        if (hFind == INVALID_HANDLE_VALUE)
-        {
-            Error = GetLastError();
-            if ((Error != ERROR_DIRECTORY) && (Error != ERROR_SHARING_VIOLATION)
-                  && (Error != ERROR_FILE_NOT_FOUND))
-            {
-                ErrorMessage(Error, pszFile);
-            }
-            return FALSE;
-        }
-
-        do
-        {
-            if (!(findData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY))
-                continue;
-
-            if (!wcscmp(findData.cFileName, L".") ||
-                !wcscmp(findData.cFileName, L".."))
-            {
-                continue;
-            }
+    DWORD dwAttributes = pFindData->dwFileAttributes;
+
```

Comment:
extra empty line!

Reply:
? You mean the line between dwAttributes and the UNREFERENCED_PARAMETER thing? well yes of course there is an empty line there, since UNREFERENCED_PARAMETER is not part of the list of local variables...
(dunno why there was even a need to explain this triviality)